src/test/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/AllocationActionIT.kt

```diff
@@ -0,0 +1,207 @@
+            defaultState = states[0].name,
+            states = states
+        )
+        createPolicy(policy, policyID)
+        createIndex(indexName, policyID)
+        val managedIndexConfig = getExistingManagedIndexConfig(indexName)
+
+        // Change the start time so the job will trigger in 2 seconds.
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+
+        waitFor { assertEquals(policyID, getExplainManagedIndexMetaData(indexName).policyID) }
+
+        // Need to wait two cycles.
+        // Change the start time so the job will trigger in 2 seconds.
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+
+        waitFor {
+            val settings = getFlatSettings(indexName)
+            assertTrue(settings.containsKey("index.routing.allocation.require.box_type"))
+            assertEquals(actionConfig.require["box_type"], settings["index.routing.allocation.require.box_type"])
+        }
+    }
+
+    fun `test allocate require`() {
+        Assume.assumeTrue(isMultiNode)
+
+        val indexName = "${testIndexName}_multinode_require"
+        val policyID = "${testIndexName}_multinode_require"
+        val actionConfig = AllocationActionConfig(require = mapOf("_name" to "integTest-1"), exclude = emptyMap(), include = emptyMap(), index = 0)
+        val states = listOf(
+                State("Allocate", listOf(actionConfig), listOf())
+        )
+
+        val policy = Policy(
+                id = policyID,
+                description = "$testIndexName description",
+                schemaVersion = 1L,
+                lastUpdatedTime = Instant.now().truncatedTo(ChronoUnit.MILLIS),
+                errorNotification = randomErrorNotification(),
+                defaultState = states[0].name,
+                states = states
+        )
+        createPolicy(policy, policyID)
+        createIndex(indexName, policyID, null, "0")
+        val managedIndexConfig = getExistingManagedIndexConfig(indexName)
+
+        // Change the start time so the job will trigger in 2 seconds.
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+
+        waitFor { assertEquals(policyID, getExplainManagedIndexMetaData(indexName).policyID) }
+
+        // Need to wait two cycles.
+        // Change the start time so the job will trigger in 2 seconds.
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+
+        waitFor {
+            val settings = getFlatSettings(indexName)
+            assertTrue(settings.containsKey("index.routing.allocation.require._name"))
+            assertEquals(actionConfig.require["_name"], settings["index.routing.allocation.require._name"])
+        }
+
+        // Third execution: Waits for allocation to complete, which will happen in this execution since index is small
+        updateManagedIndexConfigStartTime(managedIndexConfig)
+
+        waitFor {
+            assertEquals("integTest-1", getIndexShardNodes(indexName)[0])
```

Comment:
It's possible this started out on integTest-1 to begin with. I think we need to get the current value of the node its on and then migrate it to the other one and assert it or perhaps have two Allocation actions on the policy that swap it between both nodes and then add another updateManagedIndexConfigStartTime and assertion to show it was on both when required.

Reply:
Done, I've added a check to see on which node index is allocated. I think it's a cleaner solution than allocating multiple times.